File: scala/src/main/scala/com/laamella/code_state_machine/condition/StatesInactiveCondition.scala
package com.laamella.code_state_machine.condition

import com.laamella.code_state_machine.StateMachine

/**
 * This condition is met when all states passed in the constructor are active.
 */
final class StatesInactiveCondition[T, E, P <: Ordered[P]](stateMachine: StateMachine[T, E, P], statesThatMustBeInactive: T*) extends NonEventBasedCondition[E] {
  override def isMet: Boolean = {
    // TODO there is a better way to express this
    for (stateThatMustBeInactive <- statesThatMustBeInactive) {
      if (stateMachine.active(stateThatMustBeInactive)) {
        return false
      }
    }
    true
  }
}

File: java/src/main/java/com/laamella/code_state_machine/condition/StatesInactiveCondition.java
package com.laamella.code_state_machine.condition;

import com.laamella.code_state_machine.StateMachine;

/**
 * This condition is met when all states passed in the constructor are active.
 */
public final class StatesInactiveCondition<T, E, P extends Comparable<P>> extends NonEventBasedCondition<E> {
    private final T[] statesThatMustBeInactive;
    private final StateMachine<T, E, P> stateMachine;

    @SafeVarargs
    public StatesInactiveCondition(StateMachine<T, E, P> stateMachine, T... statesThatMustBeInactive) {
        this.stateMachine = stateMachine;
        this.statesThatMustBeInactive = statesThatMustBeInactive;
    }

    @Override
    public boolean isMet() {
        for (var stateThatMustBeInactive : statesThatMustBeInactive) {
            if (stateMachine.getActiveStates().contains(stateThatMustBeInactive)) {
                return false;
            }
        }
        return true;
    }
}